app/controllers/wpcc/your_results_controller.rb

```diff
@@ -3,14 +3,23 @@ class YourResultsController < EngineController
     def index
       @schedule = Wpcc::Presenter.new(schedule, view_context: view_context)
       @message_presenter = message_presenter
+      @period_percents = present_period_contribution_calculations(
+        period_percents
+      )
     end
 
     private
 
     def schedule
-      present(
-        Wpcc::ContributionsCalendar.new(contributions_params).schedule
-      )
+      present_period_contributions(calendar.schedule)
+    end
+
+    def period_percents
+      calendar.period_percents
+    end
```

Comment:
Can we not make this methods public and a helper method for the views? Then you don't need to call on index action.